extensions/jdbc/jdbc-h2/runtime/pom.xml

```diff
@@ -21,10 +21,19 @@
             <groupId>com.h2database</groupId>
             <artifactId>h2</artifactId>
         </dependency>
+        <dependency>
+            <!--
+                Needed by GraalVM to delete org.h2.fulltext.FullTextLucene
+                https://github.com/quarkusio/quarkus/issues/42228
+            -->
+            <groupId>org.apache.lucene</groupId>
+            <artifactId>lucene-core</artifactId>
+            <version>9.7.0</version>
```

Comment:
Not very excited about this one :).
We will pay the cost of having Lucene around in dev mode for instance for no particular use.

Why does it suddenly start to become an issue? More correctness checks in GraalVM?

Reply:
> Why does it suddenly start to become an issue? More correctness checks in GraalVM?

No, the issue was introduced by Quarkus with https://github.com/quarkusio/quarkus/pull/35296 in https://github.com/quarkusio/quarkus/commit/aa83e46d736fb9bb8fc2c3dafb53abdbc04fc106#diff-108c29091324b688e73c3a85be976cceb847e1dd5a8afe7968389af4a1f186f2 but gets only triggered when setting `quarkus.native.report-errors-at-runtime`.

When using this flag, GraalVM goes through all the elements of deleted classes and substitutes them so that it will throw a runtime exception when they get reached at run time. To do so it needs to load the classes (and their dependencies) which leads to this issue (see https://github.com/oracle/graal/blob/dbc87425f59a18a60aeb5c901132ba62585c09e0/substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/substitute/AnnotationSubstitutionProcessor.java#L637-L653).

Reply:
What I'm wondering is if we want to make the JVM behavior a bit worse for a GraalVM option we don't really support?

The reason I'm saying that is that in test and dev mode (and you will use H2 then), we keep in memory the jar ZIP headers and for something as big as lucene-core, it will have some cost we avoided until now.

Now maybe it's worth it, I don't know but my recollection is that we were strongly recommending against using this option so not sure we want to make configurations we support a bit worse.

Now, we probably have more and more people using containers for testing so maybe we don't really care.

Let me know what you think and if this changes your perspective. If not, I'll merge it. Thanks!

Reply:
Yeah my understanding is that it came with a lot of issues.

Let's merge and we'll adjust if some people complain.